bitcoin/src/blockdata/transaction.rs

```diff
@@ -990,6 +995,24 @@ impl Transaction {
     }
 }
 
+impl fmt::Display for Transaction {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::LowerHex::fmt(self, f) }
```

Comment:
So this gives us `.to_string()`. Is `to_hex` really that important then? Just document that `to_string()` does the same thing?

And if we want it for clarity, I'd just call `to_string()` from `to_hex()`.

Reply:
I do think we want it for clarity. But +1 to just passing through to `to_string`.

Reply:
Alright, we are making progress! This is easy to put into policy - every type that prints hex in its `Display` implementation also gets an inherent method `to_hex` that calls through to `to_string`.

Reply:
> that calls through to `to_string`

Only for types with unknown encoded lengths (e.g. all consensus types containing a `Vec`). For known-length types (e.g. hashes) `to_hex` can be faster by allocating upfront. OK, now this is a much better reason to have `to_hex` and even recommend it for use when a `String` is **really** needed.

Reply:
This thread should probably be turned into an issue but I failed to write a good one so flagging it instead.